src/main/java/org/assertj/core/internal/Failures.java

```diff
@@ -152,17 +154,35 @@ public AssertionError failureIfErrorMessageIsOverridden(AssertionInfo info) {
   /**
    * Creates a <code>{@link AssertionError}</code> using the given {@code String} as message.
    * <p>
-   * It filters the AssertionError stack trace be default, to have full stack trace use
+   * It filters the AssertionError stack trace by default, to have full stack trace use
    * {@link #setRemoveAssertJRelatedElementsFromStackTrace(boolean)}.
    *
    * @param message the message of the {@code AssertionError} to create.
    * @return the created <code>{@link AssertionError}</code>.
    */
   public AssertionError failure(String message) {
-    AssertionError assertionError = new AssertionError(message);
-    removeAssertJRelatedElementsFromStackTraceIfNeeded(assertionError);
+    return prettyPrintedWithoutAssertJRelatedElementsFromStackTraceIfNeeded(new AssertionError(message));
```

Comment:
Call  `assertionErrorCreator.assertionError(String)` instead of building directly an `AssertionError`.
Basically `AssertionError` should only be built by `AssertionErrorCreator`